src/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionStringBuilder.cs

```diff
@@ -204,9 +203,9 @@ protected internal override Expression VisitBinary(BinaryExpression node)
             if (node.NodeType == ExpressionType.ArrayIndex)
             {
                 Visit(node.Left);
-                Out("[");
+                Out('[');
```

Comment:
This change is a very low-hanging fruit. I think it is an evidence that original implementation was never seriously considered as perf-sensitive.
I am just curious - are you running into perf issues with ToString or this is just driven by a code hygiene?